app/src/main/kotlin/no/nav/arbeidsgiver/notifikasjon/infrastruktur/FunkyCache.kt

```diff
@@ -27,7 +27,7 @@ class FunkyCache<K: Any, V>(maxCapacity : Int, val loader: suspend (K) -> V) {
             }
         }
 
-    fun put(key: K, value: V) {
+    suspend fun put(key: K, value: V) = mutexes.withLock(key) {
```

Comment:
fiksen fungerer uten denne endringen

Reply:
Vi må enten fjerne locken her eller endre observeren i altinn-klienten til async. Hvis `put` kan blokkere, så vil hentingen av tilganger fra altinn bli blokkert hvis man samtidig henter virksomhetsnavn via enhetsregisteret på samme orgnr.

Siden verdien som `put`-es skal være den samme som `loader`-en returnerer, så har det ikke noe å si hvilket resultat som ender i cachet. Kjører `loader`-en allerede, så overskriver den bare det som ble `put`-et.

Selveste `cache`-mappet tåler concurrent oppdatering. Tenker at `mutexes` er for å forhindre flere kjøringer av `loader`-en, siden det er en operasjon (api-kall) vi ønsker å kjøre færrest mulig ganger. Når vi `put`-er, så er det ikke noen api-kall, og det er bare å putte den rett inn.